homeassistant/components/modern_forms/light.py

```diff
@@ -0,0 +1,144 @@
+"""Support for Modern Forms Fan lights."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiomodernforms.const import LIGHT_POWER_OFF, LIGHT_POWER_ON
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    COLOR_MODE_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+import homeassistant.helpers.entity_platform as entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from . import (
+    ModernFormsDataUpdateCoordinator,
+    ModernFormsDeviceEntity,
+    modernforms_exception_handler,
+)
+from .const import (
+    ATTR_SLEEP_TIME,
+    CLEAR_TIMER,
+    DOMAIN,
+    OPT_BRIGHTNESS,
+    OPT_ON,
+    SERVICE_CLEAR_LIGHT_SLEEP_TIMER,
+    SERVICE_SET_LIGHT_SLEEP_TIMER,
+)
+
+BRIGHTNESS_RANGE = (1, 255)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Modern Forms platform from config entry."""
+
+    coordinator: ModernFormsDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    # if no light unit installed no light entity
+    if not coordinator.data.info.light_type:
+        return
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_LIGHT_SLEEP_TIMER,
+        {
+            vol.Required(ATTR_SLEEP_TIME): vol.All(
+                vol.Coerce(int), vol.Range(min=1, max=1440)
+            ),
+        },
+        "async_set_light_sleep_timer",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_CLEAR_LIGHT_SLEEP_TIMER,
+        {},
+        "async_clear_light_sleep_timer",
+    )
+
+    async_add_entities(
+        [
+            ModernFormsLightEntity(
+                entry_id=config_entry.entry_id, coordinator=coordinator
+            )
+        ]
+    )
+
+
+class ModernFormsLightEntity(LightEntity, ModernFormsDeviceEntity):
```

Comment:
It's safer to put the integration specific base class first in the MRO.

Reply:
cleanup in #52058